app/models/manageiq/providers/cloud_manager/orchestration_stack.rb

```diff
@@ -16,4 +19,13 @@ def self.create_stack(orchestration_manager, stack_name, template, options = {})
                  :cloud_tenant           => tenant,
                  :orchestration_template => template)
   end
+
+  def self.orchestration_stack_class_factory(orchestration_manager, template)
+    case template
+    when OrchestrationTemplateVnfd
+      orchestration_manager.class::Vnf
+    else
+      orchestration_manager.class::OrchestrationStack
+    end
```

Comment:
I like the idea, especially the first part. We might name the method `#stack_type` and compose a fully qualified class name here.